src/Server/Resource/DoctrineResource.php

```diff
@@ -330,7 +332,8 @@ public function create($data)
             return $data;
         }
 
-        $entity = new $entityClass;
+        $entity = (null !== $entityFactory) ? $entityFactory->get($entityClass) : new $entityClass;
```

Comment:
@gsomoza is trying to use `__construct` with a factory in a default way for `zf-apigility-doctrine`(zad).  Currently `zad` creates an instance of an entity with raw PHP `$entity = new Entity();`

He's trying to create a factory which is robust enough to use in `zad`.  I think the idea is ok as long as the factory supports native objects.  However the library's tests appear thin and certainly there is no documentation.

I still like the idea but the cube library needs maturity.